src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -33,10 +34,30 @@ namespace wr
 		m_mipmapping_cpu_handle = d3d12::GetCPUHandle(m_mipmapping_heap, 0);
 		m_mipmapping_gpu_handle = d3d12::GetGPUHandle(m_mipmapping_heap, 0);
 
+		m_thread_pool = m_render_system.GetThreadPool();
+
+		m_command_lists.resize(d3d12::settings::max_threads + 1);
+
+		for (auto& list : m_command_lists)
+		{
+			list = d3d12::CreateCommandList(m_render_system.m_device, 1, wr::CmdListType::CMD_LIST_DIRECT);
```

Comment:
Could you use a copy cmd list instead for staging?

Reply:
No clue. It seems to be doing transitions though, so that might break it. I'll try it.